### ISSUE TITLE
Address Panics, expects, and unwraps in the src/compiler/typescript_to_rust/azle_vm_value_derive crate

### ISSUE THREAD

**Issue by bdemann** [demergent-labs/azle#1055]
Follow the example of
src/compiler/typescript_to_rust/generate_rust_canister.ts
when working with 
src/compiler/compile_rust_code.ts
In other words use spawnSync instead of execSync

**Comment by dansteren** [comment]
Once https://github.com/demergent-labs/azle/issues/1016 is merged, there will only be a few expects and panics in the codebase, all of them in the derive crate. So, we can consider this closed because all the runtime level things have been handled, or we can proceed with this ticket and add really nice error handling for these few edge cases.

It feels to me that to do this ticket properly we will need to:

- [ ] Understand how name fields in enums and structs can not have a name
- [ ] Understand the expected behavior for Unit types in Structs in Enums (as opposed to the currently supported Named and Unnamed fields)
- [ ] Remove expects and panics from the derive crate
- [ ] Add nice error messages somehow (despite being at the macro expansion phase)
- [ ] On the TS side, better handle the possibility that our `execSync` calls may error out and swap them our for `spawnSync` calls that have better error handling. Either that or this is a totally separate ticket

Given that most user-defined Structs and Enums should be handled correctly right now, I propose we move this out to the  [ 🄴 Very Very Nice Error Messages](https://github.com/demergent-labs/azle/issues/1016) epic.

**Comment by dansteren** [comment]
I didn't catch all the crazy edge runtime errors that user-defined structs and enums can have while doing #1016 so there are a few more runtime errors I'm cleaning up as part of this ticket. Just finished all of them in the try_from impl for enums. Just 3 more areas to go.
